File: hecho con python/app/controllers/analitica_controller.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.config.database import get_db
from app.dtos.analitica_dto import ResumenAcademicoDTO
from app.services.analitica_service import AnaliticaAcademicaService

router = APIRouter(prefix="/analitica", tags=["Analítica Académica"])

@router.get("/resumen", response_model=ResumenAcademicoDTO, summary="Obtener métricas y KPIs académicos globales")
def obtener_resumen(db: Session = Depends(get_db)):
    service = AnaliticaAcademicaService(db)
    return service.obtener_resumen_academico()

File: hecho con java/src/main/java/com/umanizales/notasya/controllers/AnaliticaController.java
package com.umanizales.notasya.controllers;

import com.umanizales.notasya.dtos.ResumenAcademicoDTO;
import com.umanizales.notasya.services.AnaliticaService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/analitica")
@Tag(name = "Analítica Académica", description = "Métricas y KPIs globales de rendimiento académico")
@CrossOrigin(origins = "*")
public class AnaliticaController {

    private final AnaliticaService service;

    public AnaliticaController(AnaliticaService service) {
        this.service = service;
    }

    @GetMapping("/resumen")
    @Operation(summary = "Obtener métricas y KPIs académicos globales")
    public ResponseEntity<ResumenAcademicoDTO> obtenerResumen() {
        return ResponseEntity.ok(service.obtenerResumen());
    }
}
